configuration/yaml.md

```diff
@@ -0,0 +1,642 @@
+| `channel`         | A short form for the `channels` list to define a linked channel UID for when there is only one channel with a blank config.                                                                                                                                                      |
+| `channels`        | A key-value map to define the channels linked from this item.                                                                                                                                                                                                                    |
+| `metadata`        | A key-value map to define item metadata. The metadata defined in here overrides any short forms when both are specified.                                                                                                                                                         |
+
+Example:
+
+```yaml
+version: 1
+
+items:
+  BedRoom1_Switch:
+    type: Group
+    label: Bedroom1 Light Bulb
+    groups:
+      - lBedroom1
+    tags:
+      - Lightbulb
+
+  BedRoom1_Light_Switch:
+    type: Switch
+    label: Bed Room1 Light Switch
+    icon: switch
+    format: '%s'
+    groups:
+      - BedRoom1_Light_Bulb
+    tags:
+      - Switch
+    channel: mqtt:topic:bedroom1-switch:switch1
+
+  BedRoom1_Switch_StatusLED:
+    type: Switch
+    groups:
+      - BedRoom1_Switch_Group
+    channels:
+      mqtt:topic:bedroom1-switch:statusled:
+        profile: transform:MAP
+        function: "|open=ON;closed=OFF" # | is a YAML special character, so enclose it in quotes.
+```
+
+## YAML Extensions
+
+openHAB includes custom extensions that enhance the flexibility and usability of YAML configurations.
+
+### Variable Substitutions
+
+Variable substitutions allows you to reference and reuse values dynamically within your YAML configuration files.
+This helps reduce duplication and makes templates more flexible and maintainable.
+
+#### Syntax
+
+Variables are referenced using the `${...}` syntax.
+Several forms are supported:
+
+| Syntax            | Description                                                |
+|:------------------|:-----------------------------------------------------------|
+| `${var}`          | Inserts the value of var, or an empty string if undefined. |
+| `${var-default}`  | Uses default if var is undefined.                          |
+| `${var:-default}` | Uses default if var is undefined or empty.                 |
+| `${var${nested}}` | Supports nested interpolation up to 10 levels deep.        |
+
+Notes:
+
+- Substitutions only work in double-quoted or unquoted strings.
+- Single-quoted strings are treated as literals and will not be interpolated.
+- Substitutions are also performed in key names and in default values, e.g. `${var:-${variable_default}}`.
```

Comment:
This block should be directly in Variable Substitions

Reply:
The first two entries of the list should be at line 329.
It generally describes where substitutions work

Reply:
I see your point, but I'd like to keep them as is for several reasons:
- One could view them as part of "syntax"
- Alternatively you could see the "Notes" as additional notes after having read both the introduction about Variables and the syntax.
- I'd like to keep the introduction simple as it flows onto the syntax before introducing extra complexities so that it's an easier flow to read.
- Having one "Notes" section after intro then another "Notes" after syntax makes for "Notes overload". It may be a different case if it was a huge notes that could benefit from being split up.